lib/rules/no-sparse-arrays.js

```diff
@@ -36,11 +38,25 @@ module.exports = {
         return {
 
             ArrayExpression(node) {
+                const { sourceCode } = context;
+                let commaToken;
+
+
+                for (const element of node.elements) {
+                    commaToken = sourceCode.getTokenAfter(
+                        element ?? commaToken ?? sourceCode.getFirstToken(node),
+                        astUtils.isCommaToken
+                    );
```

Comment:
When the last element does not have a trailing comma, this would unnecessarily search for a comma in the rest of the source code after the array?